lib/model/database.js

```diff
@@ -10,8 +10,18 @@ const withDatabase = (mutator) => {
   return mutator(db).finally(() => db.destroy());
 };
 
+// Same as withDatabase, but instead of assuming a knex command expects a ExplicitPromise.
+const withDatabaseAsync = (mutator) => {
```

Comment:
Ah, but they don't both return a promise. .finally is not a Promise facility, it's a knex-bluebird-Promise facility.

Reply:
Ah, I see, sounds good. Seems like a confusing method name from knex/bluebird, since it's planned for [`Promise.prototype.finally()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally) to return a `Promise`. I guess knex/bluebird maybe preceded that specification...